### DISCUSSION TITLE
how can i save metadata to Pay::Charge?

### DISCUSSION THREAD

**Discussion by ronakabhattrz** [pay-rails/pay#443] [General] [answered]
Hey @excid3 
how can I save metadata to `Pay::Charge ` i need to save order_id to metadata how can I? 

I have the following record 
`#<Pay::Charge id: 26, customer_id: 1, subscription_id: nil, processor_id: "ch_3Jahv1LPbETizUXO0Frm3v3Y", amount: 1594, currency: "usd", application_fee_amount: nil, amount_refunded: 0, metadata: {}, data: {"brand"=>"Visa", "last4"=>"4242", "exp_year"=>"2024", "exp_month"=>"4", "payment_method_type"=>"card"}, created_at: "2021-09-17 09:22:52.000000000 -0500", updated_at: "2021-09-17 09:22:55.243256000 -0500">`

Thanks in advance :)

**Comment by excid3** [comment, marked as answer]
This should work? 

```ruby
Pay::Charge.update(metadata: {order_id: 1})
```

**Reply by ronakabhattrz** [reply]
Yea right is there any method to override the object while web hook processing?

**Reply by excid3** [reply]
Also should mention you can include it during your charge.

```ruby
payment_processor.charge(19_00, metadata: {order_id: 1})
```

If you want to do processing after the charge is created, you can add a callback to the model. That's documented here: https://github.com/pay-rails/pay/blob/master/docs/8_customizing_models.md

**Reply by ronakabhattrz** [reply]
Thanks :) really helpful @excid3

**Comment by therealrodk** [comment]
If you are using Jumpstart, you can also modify the following method, which can be found in the `account` model:
```ruby
def stripe_attributes(*args)
    {
      address: billing_address&.to_stripe,
      metadata: {
        key: value
      }
    }.compact
  end
```

This is triggered and sent to Stripe as soon as someone clicks on a plan, prior to actually entering their payment information.
